packages/dev/core/src/Behaviors/Meshes/pointerDragBehavior.ts

```diff
@@ -452,7 +452,7 @@ export class PointerDragBehavior implements Behavior<AbstractMesh> {
 
                 // Project delta drag from the drag plane onto the drag axis
                 pickedPoint.subtractToRef(this.lastDragPosition, this._tmpVector);
-                dragLength = Vector3.Dot(this._tmpVector, this._worldDragAxis);
+                dragLength = Vector3.Dot(this._tmpVector, this._worldDragAxis) / (Vector3.Dot(this._worldDragAxis, this._worldDragAxis) || 1);

```

Comment:
I used the dot because this is the notation you can find in Vector projection formulas but I understand it may be more readable with lengthSquared instead.

Reply:
@CedricGuillemet Yes it could work to, let me know which solution you think is the best.

Reply:
@CedricGuillemet if I normalized it's in place, so I have to clone, I know it's minimum impact, but maybe it would be better with the lengthSquared ?

Reply:
I was still thinking about it...And now I think it's more understandable with a normalize() ^^